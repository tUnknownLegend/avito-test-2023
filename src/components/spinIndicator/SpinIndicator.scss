.spin-indicator {
  display: flex;
  height: 80vh;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  gap: var(--default-space);
}
