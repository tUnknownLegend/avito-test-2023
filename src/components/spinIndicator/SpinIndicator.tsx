import {Alert, Space, Spin} from 'antd';

function SpinIndicator() {
    return (
        <Spin tip="Loading" size="large">
            <div className="content" />
        </Spin>
    );
}

export default SpinIndicator;
