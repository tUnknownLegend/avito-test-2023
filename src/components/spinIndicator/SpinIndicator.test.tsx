import {render} from '@testing-library/react';
import SpinIndicator from './SpinIndicator.tsx';

afterEach(jest.clearAllMocks);
describe('Spin Indicator test', () => {
    it('should render correctly', () => {
        const rendered = render(<SpinIndicator/>);

        expect(rendered.asFragment()).toMatchSnapshot();
    });
});
