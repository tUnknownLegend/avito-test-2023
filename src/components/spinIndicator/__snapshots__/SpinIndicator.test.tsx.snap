// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spin Indicator test should render correctly 1`] = `
<DocumentFragment>
  <div
    class="spin-indicator"
  >
    <div
      aria-busy="true"
      aria-live="polite"
      class="ant-spin ant-spin-lg ant-spin-spinning css-dev-only-do-not-override-17a39f8"
    >
      <span
        class="ant-spin-dot ant-spin-dot-spin"
      >
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
        <i
          class="ant-spin-dot-item"
        />
      </span>
    </div>
    <h2>
      Wait for a moment...
    </h2>
  </div>
</DocumentFragment>
`;
